packages/menu-surface/index.tsx

```diff
@@ -253,62 +250,60 @@ class MenuSurface extends React.Component<MenuSurfaceProps, MenuSurfaceState> {
       getWindowScroll: () => {
         return {x: window.pageXOffset, y: window.pageYOffset};
       },
-      setPosition: (position: Position) => {
+      setPosition: (position: Partial<MDCMenuDistance>) => {
         this.setState((prevState) => Object.assign(prevState, {
           styleLeft: 'left' in position ? position.left : null,
           styleRight: 'right' in position ? position.right : null,
           styleTop: 'top' in position ? position.top : null,
           styleBottom: 'bottom' in position ? position.bottom : null,
         }));
       },
-      setMaxHeight: (maxHeight: number) => this.setState({maxHeight}),
+      setMaxHeight: (maxHeight: string) => this.setState({maxHeight}),
     };
 
-    return Object.assign(
-      {
-        addClass: (className: string) => {
-          const classList = new Set(this.state.classList);
-          classList.add(className);
-          this.setState({classList});
-        },
-        removeClass: (className: string) => {
-          const classList = new Set(this.state.classList);
-          classList.delete(className);
-          this.setState({classList});
-        },
-        hasClass: (className: string) => this.classes.split(' ').includes(className),
-        hasAnchor: () => !!this.props.anchorElement,
-        notifyOpen: () => {
-          if (this.registerWindowClickListener) {
-            this.registerWindowClickListener();
-          }
-          this.props.onOpen!();
-        },
-        notifyClose: () => {
-          if (this.deregisterWindowClickListener) {
-            this.deregisterWindowClickListener();
-          }
-          this.props.onClose!();
-        },
-        isElementInContainer: (el: HTMLElement) => {
-          if (!this.menuSurfaceElement.current) return false;
-          if (this.menuSurfaceElement.current === el) {
-            return true;
-          }
-          return this.menuSurfaceElement.current.contains(el);
-        },
-        isRtl: () => {
-          if (!this.menuSurfaceElement) return false;
-          if (!this.menuSurfaceElement.current) return false;
-          return window
-            .getComputedStyle(this.menuSurfaceElement.current)
-            .getPropertyValue('direction') === 'rtl';
-        },
-        setTransformOrigin: (transformOrigin: string) => this.setState({transformOrigin}),
+    return {
+      addClass: (className: string) => {
+        const classList = new Set(this.state.classList);
```

Comment:
if `this.state.classList` is already of Set type why do we need to instantiate this with `Set` again?

Reply:
If we just add to the existing Set, React will not detect a change meaning there won't be a re-render. React does not do a deep comparison.